Mathlib/RingTheory/Localization/Defs.lean

```diff
@@ -97,15 +97,9 @@ variable [Algebra R S] {P : Type*} [CommSemiring P]
 
 /-- The typeclass `IsLocalization (M : Submonoid R) S` where `S` is an `R`-algebra
 expresses that `S` is isomorphic to the localization of `R` at `M`. -/
-@[mk_iff]
-class IsLocalization (M : Submonoid R) (S : Type*) [CommSemiring S] [Algebra R S] : Prop where
-  /-- Everything in the image of `algebraMap` is a unit -/
-  map_units (S) : ∀ y : M, IsUnit (algebraMap R S y)
-  /-- The `algebraMap` is surjective -/
-  surj (M) : ∀ z : S, ∃ x : R × M, z * algebraMap R S x.2 = algebraMap R S x.1
-  /-- The kernel of `algebraMap` is contained in the annihilator of `M`;
-      it is then equal to the annihilator by `map_units'` -/
-  exists_of_eq : ∀ {x y}, algebraMap R S x = algebraMap R S y → ∃ c : M, ↑c * x = ↑c * y
+abbrev IsLocalization : Prop := M.IsLocalizationMap (algebraMap R S)
+
+@[deprecated (since := "2025-08-15")] alias isLocalization_iff := Submonoid.isLocalizationMap_iff
```

Comment:
I don't think this should be deprecated. I personally don't want to rely on things being "abbrev".

Reply:
It's tedious to write out ... but I added it back.
It's used exactly once in the whole library and it's more economic to replace the occurrence by `rw [IsLocalization, Submonoid.isLocalizationMap_iff]`.